File: 1721-swapping-nodes-in-a-linked-list/1721-swapping-nodes-in-a-linked-list.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode swapNodes(ListNode head, int k) {
        int temp;
        ListNode slow=head;
        ListNode fast=head;
        ListNode prev=null;
        for(int i=0;i<k;i++)
        {
            prev=fast;
            fast=fast.next;
        }
        while(fast!=null)
        {
            slow=slow.next;
            fast=fast.next;
        }
        temp=prev.val;
        prev.val=slow.val;
        slow.val=temp;
        return head;
    }
}

File: 1721-swapping-nodes-in-a-linked-list/1721-swapping-nodes-in-a-linked-list.c
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     struct ListNode *next;
 * };
 */
struct ListNode* swapNodes(struct ListNode* head, int k) {
    struct ListNode* first=head;
    for(int i=1;i<k;i++)
    {
        first=first->next;
    }
    struct ListNode* second=head;
    struct ListNode* fast=first;
    while(fast->next!=NULL)
    {
        fast=fast->next;
        second=second->next;
    }
    int temp=first->val;
    first->val=second->val;
    second->val=temp;

    return head;
}